pages/en/lb4/Decorators.md

```diff
@@ -0,0 +1,396 @@
+              '200': {
+                description: 'greeting text',
+                schema: {type: 'string'},
+              }
+            }
+          }
+        }
+      }
+    })
+    class MyController {
+      // The operation endpoint defined here will be overriden!
+      @get('/foo')
+      @param.query.number('limit')
+      greet(name) {
+      }
+    }
+    app.controller(MyController);
+  ```
+
+  A more detailed explanation can be found in [Specifying Controller APIs](controllers.htm#specifying-controller-apis)
+
+### Operation Decorator
+
+  Syntax: [`@operation(verb: string, path: string, spec?: OperationObject)`](http://apidocs.loopback.io/@loopback%2fcore/#818)
+
+  `@operation` is a controller method decorator. It exposes a Controller method as 
+  a REST API operation. You can specify the verb, path, parameters and response 
+  as specification of your endpoint, for example:
+
+  ```ts
+  const spec = {
+    parameters: [{name: 'name', type: 'string', in: 'query'}],
+    responses: {
+      '200': {
+        description: 'greeting text',
+        schema: {type: 'boolean'},
+      }
+    }
+  };
+  class MyController {
+    @operation('HEAD', '/checkExist', spec)
+    checkExist(name) {
+    }
+  }
+  ```
+  
+### Commonly-used Operation Decorators
+  
+  Syntax: [`@get(path: string, spec?: OperationObject)`](http://apidocs.loopback.io/@loopback%2fcore/#798)
+  
+  Same Syntax for decorators [`@post`](http://apidocs.loopback.io/@loopback%2fcore/#802)
+  , [`@put`](http://apidocs.loopback.io/@loopback%2fcore/#806)
+  , [`@patch`](http://apidocs.loopback.io/@loopback%2fcore/#810)
+  , [`@del`](http://apidocs.loopback.io/@loopback%2fcore/#814)
+
+  You can call these suger operation decorators as a shortcut of `@operation`, for example:
+  ```ts
+  class MyController {
+    @get('/greet', spec)
+    greet(name) {
+    }
+  }
+  ```
+
+  is equivalent to
+
+  ```ts
+  class MyController {
+    @operation('GET', '/greet', spec)
+    greet(name) {
+    }
+  }
+  ```
+  
+  For more usage, refer to [Routing to Controllers](controllers.htm#routing-to-controllers)
+
+### Parameter Decorator
+
+  Syntax: [`@param(paramSpec: ParameterObject)`]()
+
+  Optional Pattern: `@param.${in}.${type}(${name})`
+
+  `@param` can be applied to method itself or specific parameters, it describes
+  an input parameter of a Controller method.
+
+  For example:
+
+  ```ts
+    // example 1: decorator `@param ` applied on method level
+    class MyController {
+      @get('/')
+      @param(offsetSpec)
+      @param(pageSizeSpec)
+      list(offset?: number, pageSize?: number) {}
+    }
+  ```
+
+  or
+  
+  ```ts
+    // example 2: decorator `@param` applied on parameter level
+    class MyController {
+      @get('/')
+      list(
+        @param(offsetSpec) offset?: number,
+        @param(pageSizeSpec) pageSize?: number,
+      ) {}
+    }
+  ```
+
+  By the first example you apply multiple decorators to a single declaration, 
```

Comment:
`In the first example, we apply multiple decorators to a single declaration. The order in which the decorators are called is important because...`